partiql-parser/src/main/antlr/PartiQLTokens.g4

```diff
@@ -351,6 +350,34 @@ COLON: ':';
 COLON_SEMI: ';';
 QUESTION_MARK: '?';
 PERIOD: '.';
+HASH: '#';
+
+// Operators w/ special characters
+// Similar to postgresql's supported operator creation -- https://www.postgresql.org/docs/16/sql-createoperator.html
+OPERATOR
+    // may not end with + or -
+    : OpBasic+ OpBasicEnd
+    // must include at least one of OpSpecial to end w/ anything
+    | (OpBasic | OpSpecial)* OpSpecial (OpBasic | OpSpecial)*
+    ;
+
+fragment OpBasic
+    : [+*=] // TODO support `<` and `>`?
+    // comments are not matched
+    | '-' {_input.LA(1) != '-'}?
+    | '/' {_input.LA(1) != '*'}?
+    ;
+
+fragment OpBasicEnd
+    : [*/=] // TODO support `<` and `>`?
+    ;
+fragment OpSpecial
+    : [~@#%^?]  // TODO support backtick (`)?
+    // graph patterns are not matched
+    | '|' {_input.LA(1) != '!'}?
+    | '!' {_input.LA(1) != '%'}?
+    | '&' {_input.LA(1) != '%'}?
+    ;
```

Comment:
For my initial read, this looks pretty good. It covers

- Don't allow +, - at end because of number signs
- Don't allow -- or /* because comments
- Don't allow GPML patterns